Sub branch

vikasreddy-elixr opened the pull request:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

vikasreddy-elixr committed:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

vikasreddy-elixr committed:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

vikasreddy-elixr committed:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
 
         String txtfile = args[0];
```
Please check what will happen if args[0] does not contain anything. For example if user did not pass any argument what happens to the program?

vikasreddy-elixr replied:
it display the file not exists message

vikasreddy-elixr replied:
i have changed the code with dispalying the message if user does not passes any one output

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
-        checkforexistance(file);
+        checkForSupport(file);
+        checkForExistance(file);
         String searchword = args[1];
```
Please check what will happen if args[1] does not contain anything. For example if user did not pass any argument what happens to the program?

vikasreddy-elixr replied:
i have solved this with checking the args array and continue if required args are given else the execution is stopped

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
         String searchword = args[1];
         System.out.println("processing................");
         String data = null;
         data = readFileAsString(txtfile);
```
Cant we combine line number 19 and 20?

vikasreddy-elixr replied:
yeah we can do that.
resolved it.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
 
         String txtfile = args[0];
```
txtfile should be renamed in camel case. Also wouldn't it be better if we rename the txtfile to something like txtFilePath or so?

vikasreddy-elixr replied:
yeah!!
done with that..

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
         String searchword = args[1];
         System.out.println("processing................");
         String data = null;
         data = readFileAsString(txtfile);
```
What if the file is empty? If we do a check there, we can stop the execution then and there and no need to proceed further.

vikasreddy-elixr replied:
if file is empty the message is passed like the file has no data and execution will be terminated

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
         File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+        checkForSupport(file);
```
This method is a void method and if the file does not exists also it just prints "file does not exists" and program execution continues right? You might want to take care of this? Same with checkForSupport  and checkForExistence too

vikasreddy-elixr replied:
i have  used return statement to terminate the execution if it prints "file does not exists" statement

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
-        checkforsupport(file);
-        checkforexistance(file);
+        checkForSupport(file);
+        checkForExistance(file);
```
Might be good if we change the method name lie checkIfFileExists(arguments)

vikasreddy-elixr replied:
I have changed all the method names.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -12,14 +12,14 @@ public static void main(String[] args){
         data = readFileAsString(txtfile);
-        avoidspecialchar(data);
-        searchtheword(data,searchword);
+        avoidSpecialChar(data);
```
Please rename to removeSpecialCharacters

vikasreddy-elixr replied:
i have changed the method naming format for all the methods.

kssreejith1994 reviewed (changes requested):
Good work, but please make the changes suggested.

premsrj commented on src/com/company/Main.java:
```diff
@@ -36,7 +36,7 @@ public static String readFileAsString(String fileName)
     }
 
-        public static void checkforsupport(File file)
+        public static void checkForSupport(File file)
```
You should ideally be throwing an exception from here or atleast return false if the file is not in a supported format

vikasreddy-elixr replied:
i have resolved that in a  recently committed code.

premsrj commented on src/com/company/Main.java:
```diff
@@ -50,7 +50,7 @@ public static void checkforsupport(File file)
         }
 
-        public static void checkforexistance(File file) {
+        public static void checkForExistance(File file) {
```
You should ideally be throwing an exception from here or atleast return false if the file is not in a supported format

vikasreddy-elixr replied:
i have made to print as "file is not valid" if it is not in a supported format.

premsrj commented on src/com/company/Main.java:
```diff
@@ -59,10 +59,11 @@ public static void checkforexistance(File file) {
-        public static void avoidspecialchar(String data) {
+        public static void avoidSpecialChar(String data)
+        {
             data = data.replaceAll("[^a-zA-Z0-9@]", " ");
```
It is always better to return the modified data, instead of actually modifying the reference inside the method.

vikasreddy-elixr replied:
yeah i have made this change in  recently updated code.

vikasreddy-elixr committed:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        if(file.length()==0)
+        {
+            System.out.println("Enter the valid path ");
+            System.exit(0);
```
Why not just return from this method instead of explicitly calling exit() ?

vikasreddy-elixr replied:
i have replaced all the exit() statements with the return statements.

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        try {
+            searchword = args[1];
+        } catch (Exception e) {
+            System.out.println("search word has not entered");
```
```suggestion
            System.out.println("word to search was not entered");
```

vikasreddy-elixr replied:
done with that change.

vikasreddy-elixr committed:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

premsrj commented on src/com/company/Main.java:
```diff
@@ -33,43 +55,34 @@ public static String readFileAsString(String fileName)
-        public static void avoidspecialchar(String data) {
-            data = data.replaceAll("[^a-zA-Z0-9@]", " ");
+        public static boolean isFileExists(File file) {
+           return file.exists();
```
Why have a different method if it is just a one line check? You can include file.exists() directly in the if condition

vikasreddy-elixr replied:
we can do that and I have done that in recently pushed code.

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
```
What if the user did not provide any arguments?

vikasreddy-elixr replied:
it prints the the error message as "Please enter the valid path and word" and terminate the execution.

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
+        {
+            System.out.println("file is not valid");
+            System.exit(0);
```
Why not just use return statement here?

vikasreddy-elixr replied:
replaced exit() with the return statement

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        isFileFormatSupported(file);
+        isFileExists(file);
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
```
```suggestion
        if((isFileExists(file) && isFileFormatSupported(file)) == false) {
```
Proper spacing for == and { in the same line

vikasreddy-elixr replied:
I have done that.
ill check for other lines of code if there is any other issue repeated as this.

premsrj commented on src/com/company/Main.java:
```diff
@@ -33,43 +55,34 @@ public static String readFileAsString(String fileName)
+                if(st.hasMoreElements()==false)
+                {
+                    System.out.println("file is empty");
+                    System.exit(0);
```
return statement?

vikasreddy-elixr replied:
changed exit() to return statement.

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
+        if(file.length()==0)
```
```suggestion
        if(file.length()==0) {
```
curly bracket should be in the same line

vikasreddy-elixr replied:
done that for all the lines which has this scenarios.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
+        if(file.length()==0)
```
What does file.length() do? Does it validates the file path? The error message inside if statement says "enter valid path". Please check.

vikasreddy-elixr replied:
changed the print statement to  "the file does not contains any data".

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+        String txtFilePath = args[0];
```
ArrayIndexOutOfBounds exception will be thrown here if user does not pass any argument. Please check.

vikasreddy-elixr replied:
resolved that exception and updated in the recently committed code

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,49 @@
+        try {
+            txtFilePath = args[0];
+        } catch (Exception e) {
+            System.out.println("Path has not entered");
```
```suggestion
            System.out.println("Path was not entered");
```

vikasreddy-elixr replied:
changed with the println statement.

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,49 @@
+            txtFilePath = args[0];
+        } catch (Exception e) {
+            System.out.println("Path has not entered");
+            System.exit(1);
```
Return statement instead of exit() ?

vikasreddy-elixr replied:
I have replaced it with return statement.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+        isFileFormatSupported(file);
+        isFileExists(file);
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
```
You don't have to check against false again, since isFileExists and isFileFormat themselves return boolean. Please check.

vikasreddy-elixr replied:
I have changed the statement.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
+        {
+            System.out.println("file is not valid");
```
Code is duplicated here. Can we convert this to a method that takes a message as argument and prints it and exits? This way you can call the method when you want the program to exit. Check line 18 and 19 to relate this.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,18 +8,40 @@
+
+        String searchword = null;
+        try {
+            searchword = args[1];
```
Check this at the starting of the program. File path and search word are mandatory for you program to execute. So if they are not present, then you don't need to continue further.

vikasreddy-elixr replied:
yeah, I have made that with the if statement.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
                   e.printStackTrace();
               }
-              return data;
+              return data.replaceAll("[^a-zA-Z0-9@]", " ");
```
The regex pattern can be made final/constant

vikasreddy-elixr replied:
done with that.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
-                {
-                    System.out.println("file not supported");
-                }
+            return (file.getName().endsWith(".txt"));
```
Make file extension also final/constant

vikasreddy-elixr replied:
i have made the change

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -33,44 +55,34 @@ public static String readFileAsString(String fileName)
             public static void searchTheWord(String data,String searchword) {
                 StringTokenizer st = new StringTokenizer(data);
                 int count = 0;
+                if(st.hasMoreElements()==false)
```
check what does st.hasMoreElements() do

vikasreddy-elixr replied:
it was an extra code and i have removed that

kssreejith1994 reviewed (changes requested):
Please make the requested changes.

vikasreddy-elixr committed:
This is the code for the Java terminal application where it takes two command line args from the user and checks for the file through path given and search for the word and word count.

kssreejith1994 commented:
Also, please add a relatable commit message everytime.

vikasreddy-elixr committed:
I have replaced all the exit() with return statement.
modified all the Print statements and done with the scenario if the user skip passing an argument.
reduced the duplicate lines.

premsrj commented:
> Also, please add a relatable commit message everytime.

I agree, all the commits have the same message. Your commit message should have information of what was actually done in that commit.

vikasreddy-elixr committed:
I have replaced all the exit() with return statement.
modified all the Print statements and done with the scenario if the user skip passing an argument.
reduced the duplicate lines.

vikasreddy-elixr committed:
I have replaced all the exit() with return statement.
modified all the Print statements and done with the scenario if the user skip passing an argument.
reduced the duplicate lines.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,68 +8,53 @@
-        checkforsupport(file);
-        checkforexistance(file);
+        if(args.length!=2) {
+            System.out.println("Please enter the valid path and word");
```
May be you can rephrase this to "Exiting, since no file path and word to search for are provided!"

vikasreddy-elixr replied:
yeah changes done.

premsrj commented:
@vikasreddy-elixr Now you have three commits with the message 
"I have replaced all the exit() with return statement. "
Is that all that was done in the three commits?

premsrj commented on src/com/company/Main.java:
```diff
@@ -8,68 +8,53 @@
 
 
 public class Main {
```
Code seems ok now. Fix the file formatting. The indentation and spaces are all wrong. There is a menu option available in IntelliJ

Code -> Reformat File

vikasreddy-elixr replied:
done with File formatting

vikasreddy-elixr committed:
In this Code the file formatting has been fixed.

premsrj commented on src/com/company/Main.java:
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
+
+}
+
 
```
Can you remove these extra lines from 73 to 77?

vikasreddy-elixr replied:
yeah sure
I have removed it

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
         isFileFormatSupported(file);
 
-        if(!file.exists() || !isFileFormatSupported(file)) {
+        if (!file.exists() || !isFileFormatSupported(file)) {
```
isFileFormatSupported is called in line 22 as well?

vikasreddy-elixr replied:
it was duplicated,
but now the duplicated code has been removed.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
 
-        if(!file.exists() || !isFileFormatSupported(file)) {
+        if (!file.exists() || !isFileFormatSupported(file)) {
             System.out.println("file is not valid");
```
Start all the sysout messages with capital letters and end with proper punctuations.

vikasreddy-elixr replied:
done with that.

vikasreddy-elixr committed:
Removed Extra lines of codes in this code.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
-                while (st.hasMoreTokens())
-                {
+    public static boolean isFileFormatSupported(File file) {
+        return (file.getName().endsWith(".txt"));
```
Instead of hardcoding .txt, you could probably make this is a final variable?

vikasreddy-elixr replied:
created the variable and made that as you suggested

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -20,53 +21,55 @@ public static void main(String[] args) {
-        public static void searchTheWord(String data,String searchword) {
-                StringTokenizer st = new StringTokenizer(data);
-                int count = 0;
+        return data.replaceAll("[^a-zA-Z0-9@]", " ");
```
Instead of hardcoding regex here, you could probably make this is a final variable?

vikasreddy-elixr replied:
created the variable and made that as you suggested.

kssreejith1994 reviewed (changes requested):
Code looks okay, few minor comments to check.

vikasreddy-elixr committed:
Made changes in the print statements.created variables for ".txt" and "[^a-zA-Z0-9@]" instead of using those directly.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,82 +8,60 @@
-        checkforsupport(file);
-        checkforexistance(file);
+    final static String extention=".txt";
+    final static String specialChars="[^a-zA-Z0-9@]";
```
constants should be defined in capitals and should be meaningful. specialChars does not serve the purpose of the regex used.

vikasreddy-elixr replied:
done with the changes

vikasreddy-elixr committed:
Changed the name of the constants declared in the format.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,82 +8,60 @@
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
```
There is a possibility of null pointer exception here. What happens if there is no data in file and program execution continues?

vikasreddy-elixr replied:
resolved with writing replaceall method within try block

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -8,82 +8,60 @@
-
-
+    }
+}
```
Add a new line here.

vikasreddy-elixr committed:
altered the code to avoid the possibilities of null pointer exception in line 44,45.

vikasreddy-elixr committed:
altered the code such as it should also consider jason file aas argument.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -4,86 +4,67 @@
-        File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+    final static String FILE_TYPE_1=".txt";
```
Make it FILE_TYPE_TXT please

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -4,86 +4,67 @@
-        checkforsupport(file);
-        checkforexistance(file);
+    final static String FILE_TYPE_1=".txt";
+    final static String FILE_TYPE_2=".jason";
```
Make it FILE_TYPE_JSON please. and its json.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -4,86 +4,67 @@
+        if(data!=null) {
+            return data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
+        }
+        return data;
```
Data will still be null. Please do the null check in main.

vikasreddy-elixr committed:
Constants name has been changed to meaningful names.
null check statements has been added in the main method.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -1,89 +1,72 @@
+        System.out.println("Processing................");
+        String data = readFileAsString(txtFilePath);
+        if (data == null) {
+            return;
```
Add a sysout mentioning that since no data present in file,program is exiting

vikasreddy-elixr committed:
Added a sysout mentioning that since no data present in file,program is exiting.

vikasreddy-elixr committed:
replaceall has been written in try block to manage null pointer exception.
resolved the null check statements.

kssreejith1994 commented on src/com/company/Main.java:
```diff
@@ -1,89 +1,70 @@
+        String data = null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+            data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
```
Shouldn't it be data=data.replaceAll() ? Before changing to this try to read a file that contains ',' in the text and see the output.

vikasreddy-elixr replied:
yeah checked that
data replaced was not storing in the variable data

vikasreddy-elixr committed:
assigned data.replaceall method to the data variable.

vikasreddy-elixr committed:
Checked for content in the data after readFileAsString method
and updated it.

kssreejith1994 reviewed (approved):
Code looks fine now.

kssreejith1994 commented:
Merged code to master.
